### ISSUE TITLE
How to acquire the result for symbolic execution?

### ISSUE THREAD

**Issue by zhenyu-zhou** [facebook/infer#476]
Hello,

I intend to look into the control graph (or the execution path tree) provided by Infer but I find that I cannot understand the output. Ideally, I think the execution tree should look like:

a > 5
(true) b < 3
          (true) foo1
          (false) foo2
(false) foo3

Follow the instruction here: http://fbinfer.com/docs/advanced-features.html#debug-mode, I use the command `infer --debug -- <build command>` to generate the HTML files at captured/, which is said to contain the "symbolic execution result". However, I find it is extremely hard to parse the result there into the human-readable execution tree. I find each node has "preds" and "succs", which might be the edge of the execution tree. Unfortunately, when I click into a node, the information is extremely heavy and I have no idea how to translate it into the path condition (`a>5` in the example above) when a branch occurs.

Any idea to extract such information from the HTML file? I believe it should somehow be there.

Thanks in advance!

**Comment by sblackshear** [comment]
The `icfg.dot` files mentioned in the debug mode documentation might be more along the lines of what you are looking for. You can view the CFG in these files using `dotty` or any other tool that understands the graphviz format. This won't have details about the abstract state at each program point, but it should be easier to understand the flow.

**Comment by zhenyu-zhou** [comment]
Thank you! I've taken a look at `icfg.dot` and it seems to still include the abstract states“ (Please let me know if the attachment is not visible):
![screen shot 2016-10-08 at 3 58 19 pm](https://cloud.githubusercontent.com/assets/9327647/19216465/a5a0e448-8d71-11e6-901d-d4727093f5db.jpg)

I only find this one branch in my target function (in fact there should be several branches). And I'm kind of confused what the path condition is for this branch - I see both false branch (showed in the figure) and true branch (not showed in the figure, at the right arrow) but I indeed cannot figure out what causes the branch. In other words, which variable is symbolic? What is the value for it to introduce branch?

Do I miss anything here for the `icfg.dot`?

Thanks again for your help!

**Comment by jberdine** [comment]
I think that there might be some confusion here about what information `icfg.dot` is presenting.  The `icfg.dot` files contain the static control-flow graph, which is determined by the source code and control-flow paths that might be impossible are not removed.

From the comments, with the mention of e.g. "which variable is symbolic", it seems that @zzy assumes that infer operates in a way similar to King-style symbolic execution.  I don't think that trying to understand what infer does in these terms is useful, since what it actually does is very different.  A better starting point is probably http://dl.acm.org/citation.cfm?doid=2049697.2049700.

**Comment by zhenyu-zhou** [comment]
@jberdine  Thank you! I spend a while to understand the concept here and you are right, it is very different from the definition in my mind. Sorry for my misunderstanding!

**Comment by jberdine** [comment]
Closing this as discussion seems to have settled, but please reopen if there are additional questions.
